service/history/historyEventNotifier.go

```diff
@@ -144,15 +155,19 @@ func (notifier *historyEventNotifierImpl) UnwatchHistoryEvent(domainID string,
 func (notifier *historyEventNotifierImpl) dispatchHistoryEventNotification(event *historyEventNotification) {
 	identifier := &event.workflowIdentifier
 
-	notifier.Lock()
-	defer notifier.Unlock()
+	shardID := notifier.workflowIDToShardID(identifier.workflowID)
+	lock, eventsPubsubs := notifier.createOrGetShards(shardID)
+	lock.RLock()
+	defer lock.RUnlock()
 
-	eventsPubsubs, ok := notifier.eventsPubsubs[*identifier]
+	subscribers, ok := eventsPubsubs[*identifier]
 	if !ok {
 		return
 	}
 
-	for _, channel := range eventsPubsubs {
+	timer := notifier.metrics.StartTimer(metrics.HistoryEventNotificationFanoutLatency, metrics.HistoryEventNotificationFanoutLatency)
```

Comment:
First parameter needs to be the metric scope.

Reply:
typo, fixed